src/Workspaces/Core/Portable/Options/OptionSet.cs

```diff
@@ -55,6 +67,47 @@ public OptionSet WithChangedOption<T>(PerLanguageOption<T> option, string? langu
             return WithChangedOption(new OptionKey(option, language), value);
         }
 
+        internal AnalyzerConfigOptions AsAnalyzerConfigOptions(IOptionService optionService, string? language)
+        {
+            return ImmutableInterlocked.GetOrAdd(
+                ref _lazyAnalyzerConfigOptions,
+                language ?? NoLanguageSentinel,
+                (string language, (OptionSet self, IOptionService optionService) arg) => arg.self.CreateAnalyzerConfigOptions(arg.optionService, (object)language == NoLanguageSentinel ? null : language),
+                (this, optionService));
+        }
+
         internal abstract IEnumerable<OptionKey> GetChangedOptions(OptionSet optionSet);
+
+        internal virtual AnalyzerConfigOptions CreateAnalyzerConfigOptions(IOptionService optionService, string? language)
```

Comment:
Can we move `AnalyzerConfigOptionSet` down to the workspaces layer and make this a private protected method?

Reply:
We can make it private protected even if we don't move it.